frontend/src/Store/Actions/Settings/qualityDefinitions.js

```diff
@@ -44,15 +44,45 @@ export default {
     items: [],
     isSaving: false,
     saveError: null,
-    pendingChanges: {}
+    pendingChanges: {},
+
+    isLimitsFetching: false,
+    isLimitsPopulated: false,
+    limitsError: null,
+    limits: { minLimit: 0, maxLimit: 0, preferredLimit: 0 }
   },
 
   //
   // Action Handlers
 
   actionHandlers: {
     [FETCH_QUALITY_DEFINITIONS]: createFetchHandler(section, '/qualitydefinition'),
-    [SAVE_QUALITY_DEFINITIONS]: createSaveHandler(section, '/qualitydefinition'),
+
+    [FETCH_QUALITY_DEFINITION_LIMITS]: function(getState, payload, dispatch) {
+      dispatch(set({ section, isLimitsFetching: true }));
+      const promise = createAjaxRequest({
+        url: '/qualitydefinition/limits'
+      }).request;
+
+      promise.done((data) => {
+        dispatch(set({
+          section,
+          isLimitsFetching: false,
+          isLimitsPopulated: true,
+          limitsError: null,
+          limits: data
+        }));
+      });
+
+      promise.fail((xhr) => {
+        dispatch(set({
```

Comment:
If this fails to fetch we treat it as populated, but anything trying to use the limits will fail.

Reply:
What is the corrective action you'd like me to take here?

Reply:
Either the definitions shouldn't be displayed and and error message is shown instead or the limits on the server side need to be the defaults here, which really defeats the purpose.

Reply:
I have reverted all frontend changes and this will be resolved in my next push.